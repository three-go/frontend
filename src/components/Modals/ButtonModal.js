import React from "react";

import { Modal, StyleSheet, View, Text } from "react-native";
import PropTypes from "prop-types";

import { colors } from "../../common";

const ButtonModal = ({ content, isVisible, children }) => {
  const count = React.Children.count(children);

  return (
    <Modal animationType="fade" transparent={true} visible={isVisible}>
      <View style={styles.container}>
        <View style={styles.modalContainer}>
          <Text style={styles.textContainer}>{content}</Text>
          <View style={styles.buttonContainer(count)}>{children}</View>
        </View>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 22,
  },
  modalContainer: {
    justifyContent: "center",
    alignItems: "center",
    width: "75%",
    height: "25%",
    borderWidth: 5,
    borderRadius: 20,
    borderColor: colors.ivory,
    backgroundColor: colors.gray,
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  textContainer: {
    marginBottom: 20,
    color: colors.ivory,
    fontSize: 18,
    fontWeight: "bold",
  },
  buttonContainer: (count) => {
    return {
      flexDirection: "row",
      justifyContent: count > 1 ? "space-evenly" : "center",
      alignItems: "center",
      width: "100%",
    };
  },
});

export default ButtonModal;

ButtonModal.propTypes = {
  content: PropTypes.string.isRequired,
  isVisible: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
};
