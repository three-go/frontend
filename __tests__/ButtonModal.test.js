import React from "react";
import { render } from "@testing-library/react-native";
import "@testing-library/react-native";
import { Text } from "react-native";
import ButtonModal from "../src/components/Modals/ButtonModal";

it("ButtonModal should render correctly", () => {
  const component = render(
    <ButtonModal content="test" isVisible={true}>
      <Text />
    </ButtonModal>
  );

  expect(component).toBeTruthy();
  expect(component).toMatchSnapshot();

  const { getByText } = render(
    <ButtonModal content="test" isVisible={false}>
      <Text />
    </ButtonModal>
  );

  expect(() => getByText(/test/i)).toThrow(
    "Unable to find an element with text: /test/i"
  );
});

it("ButtonModal should render content text", () => {
  const { getByText, toJSON } = render(
    <ButtonModal content="test" isVisible={true}>
      <Text />
    </ButtonModal>
  );
  const buttonModal = toJSON();

  expect(getByText(/test/i)).toHaveTextContent("test");
  expect(buttonModal).toBeTruthy();
  expect(buttonModal).toMatchSnapshot();
});

it("ButtonModal should render children correctly", () => {
  const { getByText, toJSON } = render(
    <ButtonModal content="test" isVisible={true}>
      <Text>children rendered</Text>
    </ButtonModal>
  );
  const buttonModal = toJSON();

  expect(getByText(/children rendered/i)).toHaveTextContent(
    "children rendered"
  );
  expect(buttonModal).toBeTruthy();
  expect(buttonModal).toMatchSnapshot();
});
