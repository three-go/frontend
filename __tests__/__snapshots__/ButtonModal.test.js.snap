// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonModal should render children correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 22,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#96A1A8",
          "borderColor": "#FCF8F6",
          "borderRadius": 20,
          "borderWidth": 5,
          "elevation": 5,
          "height": "25%",
          "justifyContent": "center",
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "width": "75%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FCF8F6",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 20,
          }
        }
      >
        test
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text>
          children rendered
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`ButtonModal should render content text 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 22,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#96A1A8",
          "borderColor": "#FCF8F6",
          "borderRadius": 20,
          "borderWidth": 5,
          "elevation": 5,
          "height": "25%",
          "justifyContent": "center",
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "width": "75%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FCF8F6",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 20,
          }
        }
      >
        test
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
</Modal>
`;

exports[`ButtonModal should render correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 22,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#96A1A8",
          "borderColor": "#FCF8F6",
          "borderRadius": 20,
          "borderWidth": 5,
          "elevation": 5,
          "height": "25%",
          "justifyContent": "center",
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "width": "75%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FCF8F6",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 20,
          }
        }
      >
        test
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
</Modal>
`;
